crates/bindings/src/lib.rs

```diff
@@ -1030,6 +1032,50 @@ impl ReducerContext {
     pub fn as_read_only(&self) -> ViewContext {
         ViewContext::new(self.sender)
     }
+
+    ///  Create a new UUIDv4 using the built-in RNG.
+    //// # Example
+    /// ```no_run
+    /// # #[cfg(target_arch = "wasm32")] mod demo {
+    /// use spacetimedb::{reducer, ReducerContext, Uuid};
+    ///
+    /// #[reducer]
+    /// fn generate_uuid_v4(ctx: &ReducerContext) -> Uuid {
+    ///     ctx.new_uuid_v4()
+    /// }
+    /// # }
+    /// ```
+    pub fn new_uuid_v4(&self) -> anyhow::Result<Uuid> {
+        let mut bytes = [0u8; 16];
+        self.rng().try_fill_bytes(&mut bytes)?;
+        Ok(Uuid::from_random_bytes_v4(bytes))
+    }
```

Comment:
```suggestion
    ///  Create a new UUIDv4 using the built-in RNG.
    //// # Example
    /// ```no_run
    /// # #[cfg(target_arch = "wasm32")] mod demo {
    /// use spacetimedb::{reducer, ReducerContext, Uuid};
    ///
    /// #[reducer]
    /// fn generate_uuid_v4(ctx: &ReducerContext) -> Uuid {
    ///     ctx.new_uuid_v4()
    /// }
    /// # }
    /// ```
    #[cfg(feature = "rand")]
    pub fn new_uuid_v4(&self) -> anyhow::Result<Uuid> {
        let mut bytes = [0u8; 16];
        self.rng().try_fill_bytes(&mut bytes)?;
        Ok(Uuid::from_random_bytes_v4(bytes))
    }
```

This method needs to be feature-gated behind `rand`. If a user compiles without default features, they can't use this method.